### DISCUSSION TITLE
Proposal: roles for this collection

### DISCUSSION THREAD

**Discussion by staticdev** [nextcloud/ansible-collection-nextcloud-admin#228] [Ideas]
I just did tests with the collection.. all work file.

I was wondering we could have:

- nextcloud.admin.install (current `nextcloud.admin.install_nextcloud` is a bit redundant)
- nextcloud.admin.backup (could be based on https://github.com/aalaesar/backup_nextcloud)
- nextcloud.admin.restore -> uses the backup for previous roles to fill in a newly installed version

What do you think @aalaesar @wiktor2200 ?

**Comment by aalaesar** [comment]
Hello there !
- regarding the role's name, a short discussion took place on #171 . the argument was made by @wiktor2200 to keep the initial role's name for current user's retro-compatibility . I've found that argument sounding good and we went ahead. for now .
- the backup role I wrote is a personal take on nextcloud backup and may not follow official recommendation. it probably needs refinement.

- it will probably be a good thing to propose some backup and restore options in the collection.

FYI: For now I'm focusing on app modules developments alongside some personal projects 😃

**Comment by wiktor2200** [comment]
That's good idea, it would be Swiss-knife role for Nextcloud covering most of cases. I really like it.

Regarding backup and restore, I've contributed a little bit to @aalaesar's backup role and it works very well, probably it needs some molecule tests and re-thinking limitations and caveats for this role. For specific use cases it's great, but having it in general availability can cause us lots of unexpected issues (someone would think different or expect sth else), etc.
What I missing there is adding some encryption (preferably GPG) and option to send it somewhere to external place (such S3 bucket). If I would have some time I'll definitely work on that. :)
Right now it creates backup of DB, setting, files and can be stored on same machine or be downloaded by role (which also have some limitations) - and what comes to my mind is retention (but still it's getting more and more complicated). :)

**Comment by wiktor2200** [comment]
Hi @aalaesar @staticdev @AndyScherzinger! I'd like to bump this thread as I think name of collection is quite misleading :(
![screen](https://github.com/nextcloud/ansible-collection-nextcloud-admin/assets/29948368/8494caa2-ec69-420b-8fff-235bda225b62)
As you can see on screenshot, `admin` is not so popular when you're looking for `Nextcloud` phrase. As we provide whole solution I think we should re-name this role to be Nextcloud (as it's under official Nextcloud account on Galaxy).

What is more it would be good to have it mentioned on main nextcloud.com page here: https://nextcloud.com/install/#instructions-server under `Community projects` tab. Is it possible? @AndyScherzinger @jospoortvliet @blizzz @mejo-

**Reply by staticdev** [reply]
Following that train of though I would change to:

- nextcloud.nextcloud.install (current nextcloud.admin.install_nextcloud is a bit redundant)
- nextcloud.nextcloud.backup (could be based on https://github.com/aalaesar/backup_nextcloud)
- nextcloud.nextcloud.restore -> uses the backup for previous roles to fill in a newly installed version

Then I would also rename the role install_nextcloud to not be `nextcloud.nextcloud.install_nextcloud`. Not really a problem to rename IMO, quite simple actually.

**Reply by wiktor2200** [reply]
I agree with that, at first we wanted to keep the old name of role, to remain recognizable, but unfortunately galaxy UI doesn't help with that.

**Reply by aalaesar** [reply]
Hello there !
Finally got some  time to catch-up with you ! 😄 
First, thank you all for keeping the repository maintained.

regarding the collection name, I'm not favorable to renaming it:
The way ansible galaxy have setup their collection namespace + name FQCN is not trivial by itself and its implementation on the galaxy website is the source of confusion from my point of view. As an example, take the kubernetes collection: it's just named "core" in the kubernetes namespace.
So having `nextcloud.admin` as a name is , IMHO, a good representation of  both the objective and content of the collection. we should not take a website as a constraint to name the collection.
what we could do to improve or visibility/popularity:
 - At first, the website confusion can be mitigated by changing the collection's short description to emphasis that it's an official community project. It's an easy and painless change.
 - Second, we need to have the collection somewhere in the official doc as @wiktor2200  mentionned. We can also make some extra communication in some community channels. I'll check what can be done.
 - Third, as we are already working on it, we need to extend the content of this collection for more use cases. this will help usage grow.

By the way, search on ansible galaxy gave misleading results: it only display a count of elements in the collection that are matching the search words, that's why it it displaying 0 modules when you're searching for "nextcloud"
so if we rename the role to "install", which is good, the collection could simply disapear from the search results, as no element would contains the "nextcloud" word in it. 😞 

regards

**Reply by staticdev** [reply]
Fairly good points, I would add that in face to this study of Galaxy search engine, we should open a bug to them (seriously!). I also think that the name we choose for namespace and collection is well though, but it is super unfortunate that the search does not do that.

Having that said, I still think renaming `install_nextcloud` to `install` is worth it. It is shorter/cleaner/less redundant name and with this full-qualified name that Ansible recommends `nextcloud.admin.install` looks very readable. If you agree I create a new PR and also try to improve some search terms and metadata to have more nextcloud related stuff. What do you think @aalaesar @wiktor2200 ?

**Reply by aalaesar** [reply]
> Fairly good points, I would add that in face to this study of Galaxy search engine, we should open a bug to them (seriously!). I also think that the name we choose for namespace and collection is well though, but it is super unfortunate that the search does not do that.

Actually, I'm not sure it's a bug. This looks like more a feature request to do on our side for their search engine. not sure it's going to be proceeded. I'll try to dive into the search doc if there is any.

> Having that said, I still think renaming `install_nextcloud` to `install` is worth it. It is shorter/cleaner/less redundant name and with this full-qualified name that Ansible recommends `nextcloud.admin.install` looks very readable. If you agree I create a new PR and also try to improve some search terms and metadata to have more nextcloud related stuff. What do you think @aalaesar @wiktor2200 ?

Ok. Le'ts find how improve the collection rating on search  first then rename the role 😃.

I'll open a PR to change the short description and we can discuss on it here.
I'll open another discussion for documention on the nextcloud siteso we stay on the subject here about new content

**Reply by wiktor2200** [reply]
Hi! I think opening a ticket in Ansible search is a good idea to at least get some feedback how it works.

Changing a description cost us nothing and can be proceed quickly.

And still without having our role mentioned on official site/Github we won't get higher audience, so that's why I was mentioning people from NC organization. (I will try again, as last message got lost in inboxes ;) ) @AndyScherzinger @jospoortvliet @blizzz @mejo-

**Reply by wiktor2200** [reply]
Hi! I've modified collection's description to emphasize that it's official. Please review #304
